src/libs/utils.js

```diff
@@ -281,3 +281,20 @@ export const withBusyState = _.curry((setBusy, fn) => async (...args) => {
 export const useOnMount = fn => {
   useEffect(fn, [])
 }
+
+export const trimClustersOldestFirst = _.flow(
+  _.remove({ status: 'Deleting' }),
+  _.remove({ status: 'Error' }),
+  _.sortBy('createdDate')
+)
+
+export const handleNonRunningCluster = ({ status, googleProject, clusterName }, JupyterAjax) => {
+  switch (status) {
+    case 'Stopped':
+      return JupyterAjax.cluster(googleProject, clusterName).start()
+    case 'Creating':
+      return delay(15000)
+    default:
+      return delay(3000)
```

Comment:
FYI, while testing this I noticed that we have a parallel refresh process in `ClusterManager`, so we're actually making twice as many calls in total. Not something we should fix here, but good to keep in mind for a larger round-up of this code later.

Reply:
Legit.